encoding/wkt/wkt_encode.go

```diff
@@ -261,367 +252,317 @@ const (
-			}
-		}
-
-		err := enc.puts("MULTIPOLYGON ")
+		err := enc.string("MULTIPOLYGON ")
 		if err != nil {
 			return err
 		}
 
-		return enc.encodePolys(polys, removePointless, last)
+		return enc.encodePolys(polys, len(polys) - 1)
 
 	case geom.Collectioner:
 		var geoms []geom.Geometry
 
-		if !isNil(g) {
+		if !cmp.IsNil(g) {
 			geoms = g.Geometries()
 		}
 
-		last := -1
-		isPointless := true
-
-		for i := len(geoms) - 1; i >= 0; i-- {
-			pl, err := isPointlessGeo(geoms[i])
-			if err != nil {
-				return err
-			}
-			if !pl {
-				last = i
-				isPointless = false
-				break
-			}
-		}
-
-		if isPointless {
-			if removePointless && isCollectionItem {
-				return nil
-			} else if removePointless {
-				return enc.puts("GEOMETRYCOLLECTION EMPTY")
-			}
+		if len(geoms) == 0 {
+			return enc.string("GEOMETRYCOLLECTION EMPTY")
 		}
 
-		err := enc.puts("GEOMETRYCOLLECTION ")
+		err := enc.string("GEOMETRYCOLLECTION ")
 		if err != nil {
 			return err
 		}
 
-		err = enc.putc('(')
+		err = enc.byte('(')
 		if err != nil {
 			return err
 		}
 
+		last := len(geoms) - 1
+
 		for _, v := range geoms[:last] {
-			pl, err := isPointlessGeo(v)
-			if err != nil {
-				return err
-			}
-			if removePointless && pl {
-				continue
-			}
-			err = enc.encode(v, removePointless, true)
+			err = enc.encode(v, true)
 			if err != nil {
 				return err
 			}
 
-			err = enc.putc(',')
+			err = enc.byte(',')
 			if err != nil {
 				return err
 			}
 		}
 
-		pl, err := isPointlessGeo(geoms[last])
-		if err != nil {
-			return err
-		}
-		if removePointless && pl {
-			panic("the last element must always be poinfull, set by caller")
-		}
-
-		err = enc.encode(geoms[last], removePointless, true)
+		err = enc.encode(geoms[last], true)
 		if err != nil {
 			return err
 		}
 
-		return enc.putc(')')
+		return enc.byte(')')
 
 	// non basic types
 
 	case geom.Line:
-		return enc.encode(geom.LineString(g[:]), removePointless, false)
+		return enc.encode(geom.LineString(g[:]), false)
 
 	case [2][2]float64:
-		return enc.encode(geom.LineString(g[:]), removePointless, false)
+		return enc.encode(geom.LineString(g[:]), false)
 
 	case [][2]float64:
-		return enc.encode(geom.LineString(g), removePointless, false)
+		return enc.encode(geom.LineString(g), false)
 
 	case []geom.Line:
 		lines := make(geom.MultiLineString, len(g))
 		for i, v := range g {
 			lines[i] = [][2]float64(v[:])
 		}
 
-		return enc.encode(lines, removePointless, false)
+		return enc.encode(lines, false)
 
 	case []geom.Point:
 		points := make(geom.MultiPoint, len(g))
 		for i, v := range g {
 			points[i] = v
 		}
 
-		return enc.encode(points, removePointless, false)
+		return enc.encode(points, false)
 
 	case geom.Triangle:
-		return enc.encode(geom.Polygon{g[:]}, false, false)
+		return enc.encode(geom.Polygon{g[:]}, false)
 
 	case geom.Extent:
-		return enc.encode(g.AsPolygon(), false, false)
+		return enc.encode(g.AsPolygon(), false)
 
 	case *geom.Extent:
 		if g != nil {
```

Comment:
Should be add the base geom types. E.g. geom.Point, geom.Linestring, etc...

I started to feel that the interfaces were a mistake. Though I'm not sure and have been going back and forth with @ARolek  on this since the start of this project...

Reply:
Yeah, drop the interfaces. It's better to do it sooner rather than later.